### DISCUSSION TITLE
use of in-memory directory to save files

### DISCUSSION THREAD

**Discussion by mrosseel** [WimDeMeester/eFinder#18] [Ideas]
I've done some experimenting with adding a /var/tmp directory with tmpfs - this speeds up things a bit, but with added installation complexity.

Apparently there's an already running in-memory tmpfs dir called '/dev/shm' which is 4Gb on my 8Gb PI.

Proposal:
- My proposal is to use '/dev/shm' to store files to gain some speed and also spare the SD card from frequent writes.
- I'd like to remove all '/Solver' dir references and just treat the root of the cloned source code as being the base. Images will be saved to /dev/shm (configurable of course, in case it is run on something other than a pi), test images are already present in the base dir so don't need to be copied. Don't know what to do with the Stills, where can I find these test files?

**Comment by WimDeMeester** [comment]
Good plan to store things in /dev/shm.  The Stills are the images that are taking during the session (I think, I 'll check later today).  Would be a nice option to be able to store them on disk.

**Comment by WimDeMeester** [comment]
I checked the Stills folder.  This directory does indeed have the images that were platesolved in it.

**Comment by mrosseel** [comment]
indeed I've soon this copy happens during capture (not sure how much time is lost with this) but it's a good debugging tool for sure. 
So the question is:
- do we store it at ~/efinderclone/Stills or at ~/Solver/stills ?
==> my preference is in the cwd, which should be ~/efinderclone, in this way the Solver dir is no longer needed and everything can either stay in the cwd (test jpegs), be saved in-memory (/dev/shm for captures and astrometry.net files) or stored in an extra dir in the cwd. This dir can be created if not present, and ignored in .gitignore

**Comment by WimDeMeester** [comment]
I would store it in the cwd, but I would make add an option in eFinder.config to enable or disable saving the file to disk.

**Reply by mrosseel** [reply]
good idea, I'll do this in another PR

**Comment by WimDeMeester** [comment]
Is the platesolving also being done in the shared memory?
